tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +302,110 @@ def guess_realert_every(job):
     return realert_every
 
 
-def compute_check_result_for_job(client, job):
-    kwargs = {
-        "name": "check_tron_job.{}".format(job['name']),
-        "source": "tron",
+def filter_monitoring_config(config):
+    """ Filters out non-Sensu arguments from a job's monitoring config """
+    sensu_args = set([
+        'name',
+        'runbook',
+        'status',
+        'output',
+        'team',
+        'page',
+        'tip',
+        'notification_email',
+        'check_every',
+        'realert_every',
+        'alert_after',
+        'dependencies',
+        'irc_channels',
+        'slack_channels',
+        'ticket',
+        'project',
+        'source',
+        'tags',
+        'ttl',
+        'sensu_host',
+        'component',
+        'description',
+    ])
+
+    include, exclude = {}, {}
+    for k, v in config.items():
+        (include if k in sensu_args else exclude)[k] = v
+    return include, exclude
+
+
+def sort_runs_by_interval(job_content, interval='day'):
+    """ Sorts a job's runs by a time interval (year, month, day, hour, minute,
+    or second), according to a job run's run time.
+    """
+    interval_formats = {
+        'year': '%Y',
+        'month': '%Y.%m',
+        'day': '%Y.%m.%d',
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
-    if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
 
-    status = job["status"]
-    if status == "disabled":
-        kwargs["output"] = "OK: {} is disabled and won't be checked.".format(
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        # Sort runs by shortdate
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            shortdate = time.strftime(
+                interval_formats[interval],
+                _timestamp_to_timeobj(run['run_time']),
+            )
+            run_buckets[shortdate].append(run)
+    return dict(run_buckets)
+
+
+def compute_check_result_for_job(client, job):
+    kwargs = m(
+        name="check_tron_job.{}".format(job['name']),
+        source="tron",
+    )
+    if 'realert_every' not in kwargs:
+        kwargs = kwargs.set('realert_every', guess_realert_every(job))
+    kwargs = kwargs.set('check_every', "{}s".format(_run_interval))
+    sensu_kwargs, nonsensu_kwargs = filter_monitoring_config(job['monitoring'])
```

Comment:
I understand the pattern that you're going for, but there is a lot of indirection to deal with one non-standard attribute

at this point, you could just do

```
monitor_each_run = job['monitoring'].pop(PRECIOUS_JOB_ATTR, False)
kwargs.update(job['monitoring'])
....
if monitor_each_run:
    dated_runs = sort_runs_by_interval(job_content, interval='day')
...
```
and delete `filter_monitoring_config` and `_are_runs_precious`

sure, it might need reworking if the list of monitoring attributes that aren't sensu related grows, but we can tackle that when we come to it. For now I'd vote for the 'easy' option.

Reply:
I agree with the easy approach; `filter_monitoring_config` and `_are_runs_precious` are easily removable.